apps/condo/domains/notification/schema/fields/MessageType.js

```diff
@@ -0,0 +1,28 @@
+const { MESSAGE_TYPES } = require('@condo/domains/notification/constants/constants')
+
+function getMaxLength () {
+    const maxTypeLength = Math.max(...MESSAGE_TYPES.map(t => t.length))
+    let pow = 0
+    while (Math.pow(2, pow) < maxTypeLength) {
+        pow++
+    }
+
+    return Math.pow(2, pow)
+}
+
+function getMessageTypeField ({ isRequired, ...extraAttrs }) {
+    return {
+        type: 'Select',
+        options: MESSAGE_TYPES,
+        graphQLReturnType: 'MessageType',
+        isRequired,
+        kmigratorOptions: { null: !isRequired, max_length: getMaxLength() },
```

Comment:
Dynamic `max_length` means possible unplanned migration. Is it ok?

Reply:
It's not that dynamic. Its growth algorithm is similar to list capacity growth. The idea is to limit length from some power of 2. So now max_len of all constants is `73`, so the closest power of 2 to it is `128`. So it will be static (until you create some type with `> 128` symbols :)